### PR TITLE
Model File Types Support

### PR DESCRIPTION
**Pull request by inf0rmatiker**
This leaves behind the notion that all models must be a .zip file.
Models can be accepted as:
- `.pth` for PyTorch (Pickle zipfile)
- `.pickle` for Scikit-Learn (Pickle file)
- either `.zip` (SavedModel) or `.h5` (HDF5) for Tensorflow

### CODE REVIEW

File: overlay/tensorflow_validation/validation.py [hunk 1]
```diff
@@ -86,22 +86,27 @@ def validate_gis_joins_multithreaded(self, gis_joins: list) -> list:
...
-                self.label_field,
+                self.request.feature_fields,
+                self.request.label_field,
                 limit,
                 sample_rate
             )
 
             # Load MongoDB Documents into Pandas DataFrame
             features_df = pd.DataFrame(list(documents))
+
```
Comment (inf0rmatiker) [inline overlay/tensorflow_validation/validation.py:105]: Once we have the Documents in-memory as a Pandas DataFrame, we need to close the connection ASAP so the router doesn't get overwhelmed with open connections. I've noticed this happening and the router crashing if we wait to close the connection until after we evaluate the model.

File: overlay/tensorflow_validation/validation.py [hunk 2]
```diff
@@ -1,46 +1,40 @@
...
+from overlay.validation_pb2 import ValidationMetric, ValidationJobRequest, BudgetType, StaticBudget
 from overlay.db.querier import Querier
-from overlay.constants import DB_HOST, DB_PORT, DB_NAME, MODELS_DIR
+from overlay.constants import MODELS_DIR
 
 
 class TensorflowValidator:
 
     def __init__(self, request: ValidationJobRequest):
-        self.job_id = request.id
```
Comment (inf0rmatiker) [inline overlay/tensorflow_validation/validation.py:15]: No need to make duplicates of the request fields. This just makes it harder to update the protobuf request definition.

File: overlay/tensorflow_validation/validation.py [hunk 3]
```diff
@@ -1,46 +1,40 @@
...
-        self.validation_metric = request.validation_metric
-        self.normalize = request.normalize_inputs
-        self.validation_budget = request.validation_budget
+        self.request: ValidationJobRequest = request
 
     def load_tf_model(self, verbose=False):
         # Load Tensorflow model from disk
-        model_path = f"{MODELS_DIR}/{self.job_id}"
+        model_path = f"{MODELS_DIR}/{self.request.id}/"
+        first_entry = os.listdir(model_path)[0]
```
Comment (inf0rmatiker) [inline overlay/tensorflow_validation/validation.py:22]: Could be either the directory containing the `SavedModel` format, or a `.h5` HDF5 model file

File: overlay/tensorflow_validation/validation.py (hunk 2 above)
Comment (inf0rmatiker) [reply overlay/tensorflow_validation/validation.py:15]: Note-- this change is also reflected in all 3 validation packages

**Review by menuka94 — APPROVED** [review]

**Pull request merged**